core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -317,6 +320,31 @@ public UpdateSchema unionByNameWith(Schema newSchema) {
     return this;
   }
 
+  @Override
+  public UpdateSchema setIdentifierFields(Set<String> names) {
```

Comment:
I think we may want to also keep `addIdentifierField`. There's still a use case around adding a field at a time. Plus, given that I think we want to be able to specify the parent (see my comment below), this form of the method is a bit awkward because we can't pass pairs of strings, `(parent, name)`.

What about changing this to `resetIdentifierFields`, `addIdentifierField`, and `removeIdentifierField`? I think those get the same job done because `set` is `reset` plus `add` calls.

Reply:
I'm rethinking this; sorry for the churn.

We should actually be able to do this with a single string and no parent name. When `add` or `set` is called, the column must already exist. We ensure that each name is unique: `a.b` is either a field named `a.b` or is a field named `b` within field `a`, but never both. Because of that, the name here is never ambiguous. We will always be able to look it up. So supporting `Collection<String>` or `String...` should work as long as we track two things: existing identifier fields as ids and new identifier fields as names. Then when we create the identifier list later we can look up the names.

Does that sound reasonable?